ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -14,25 +14,148 @@ import ReactiveCocoa
 class SignalSpec: QuickSpec {
 	override func spec() {
 		describe("init") {
-			pending("should run the generator immediately") {
-			}
-
-			pending("should keep signal alive if not terminated") {
-			}
-
-			pending("should deallocate after erroring") {
-			}
-
-			pending("should deallocate after completing") {
-			}
-
-			pending("should forward events to observers") {
-			}
-
-			pending("should dispose of returned disposable upon error") {
-			}
-
-			pending("should dispose of returned disposable upon completion") {
+			let testError = NSError(domain: "SignalSpec", code: 1, userInfo: nil)
+			var testScheduler: TestScheduler!
+			
+			beforeEach {
+				testScheduler = TestScheduler()
+			}
+			
+			it("should run the generator immediately") {
+				var didRunGenerator = false
+				Signal<AnyObject, NoError> { observer in
+					didRunGenerator = true
+					return nil
+				}
+				
+				expect(didRunGenerator).to(beTruthy())
+			}
+
+			it("should keep signal alive if not terminated") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal { observer in
+					return nil
+				}
+				
+				expect(signal).toNot(beNil())
+			}
+
+			it("should deallocate after erroring") {
+				weak var signal: Signal<AnyObject, NSError>? = Signal { observer in
+					testScheduler.schedule {
+						sendError(observer, testError)
+					}
+					return nil
+				}
+				
+				var errored = false
+				
+				signal!.observe(error: { _ in errored = true })
```

Comment:
I wonder if I should've used optional chain here, like `signal?.observe()` since `!` would theoretically crash tests if signal happens to be `nil`. Any thoughts?